Delivered-To: pidge@toganlabs.com
Received: by 2002:a50:c091:0:0:0:0:0 with SMTP id k17-v6csp3195613edf;
        Fri, 19 Oct 2018 04:51:04 -0700 (PDT)
X-Google-Smtp-Source: ACcGV62Oo0Mk8uo0LJKakIBzx4C0N9Ard2jlevFpjDGbQNiDAeJakVnPPakNmVk3FV682guDUky3
X-Received: by 2002:a5d:4fcf:: with SMTP id h15-v6mr31205137wrw.261.1539949864553;
        Fri, 19 Oct 2018 04:51:04 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1539949864; cv=none;
        d=google.com; s=arc-20160816;
        b=cLbTZiEnVKsItNF8hGCp54u7F6fHk/EvQ1VwwY7ftiqBd6XuWE4UYTgikbuaC110/I
         c/3zOcnekTVAJ72v7jIQsvDeclB9tAoZvkNDn07/uyR/nuAzBFd3VLoTrZobq+K/29kE
         cToDGkBxezhwBM+si3ZUzzBV1vB6ukkjWPZB4SFXuJp95pby+w2vUojFppgM9kd3d7ua
         k8eMXJdRXzwldM4VmLYN2kNa4PdwWG3gJ/r5kVUxb4E3WKVOckmXk6YpgjrUrub1/roo
         grT2BzK4yovLiQHhlagH4gpWi4laRUJ4NKpVRpgFXnOrimlO+3/tR9CrJj6m61C3WOJw
         QspQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=mime-version:user-agent:message-id:in-reply-to:date:references
         :subject:cc:to:from;
        bh=MxeH28cCUZEmO4A/mByevqZDHU2yQj2gffkc1Q/kOng=;
        b=0YWzmdtFo2JuPKxFj8HeLK3y2biPPqYt3+P61U839MNZZJq9lpA4hkCxlhqLAuMFXG
         6gdznbvgv8gsPBB68laGMlqhGCTmUPourqYLRPAsq63+NuGvk3YoEhjwFbwyKpgoEZeM
         8kOXkZ3bdL802kLVZGPC83y2WMkSHaxEegGTCsSVtbgv+4vZXtriGnF+fqj+AvMJypC1
         kcAnF7dPZv5rhkGTWgNVwXtipHgRawgBnkSzbHXib0nomIQC4Jg36aWESWBDkPXiud+Z
         XcfuJO4YLJKj9s4mQyDcZR1lXm60ZQN1VKKRNQJCgiIBDtP/6IbEp4JgmnDfXkGCMF4m
         +BMw==
ARC-Authentication-Results: i=1; mx.google.com;
       spf=pass (google.com: best guess record for domain of john.ogness@linutronix.de designates 2a01:7a0:2:106d:700::1 as permitted sender) smtp.mailfrom=john.ogness@linutronix.de
Return-Path: <john.ogness@linutronix.de>
Received: from Galois.linutronix.de (Galois.linutronix.de. [2a01:7a0:2:106d:700::1])
        by mx.google.com with ESMTPS id t132-v6si2615278wmf.77.2018.10.19.04.51.04
        for <pidge@toganlabs.com>
        (version=TLS1_2 cipher=AES128-SHA bits=128/128);
        Fri, 19 Oct 2018 04:51:04 -0700 (PDT)
Received-SPF: pass (google.com: best guess record for domain of john.ogness@linutronix.de designates 2a01:7a0:2:106d:700::1 as permitted sender) client-ip=2a01:7a0:2:106d:700::1;
Authentication-Results: mx.google.com;
       spf=pass (google.com: best guess record for domain of john.ogness@linutronix.de designates 2a01:7a0:2:106d:700::1 as permitted sender) smtp.mailfrom=john.ogness@linutronix.de
Received: from localhost ([127.0.0.1] helo=vostro.local) by Galois.linutronix.de with esmtp (Exim 4.80) (envelope-from <john.ogness@linutronix.de>) id 1gDTJ1-00064F-8m; Fri, 19 Oct 2018 13:51:03 +0200
From: John Ogness <john.ogness@linutronix.de>
To: Beth Flanagan <pidge@toganlabs.com>
Cc: Behan Webster <behanw@converseincode.com>
Subject: [PATCH] pocketbeagle: activate spi, add mcp23s18
References: <874ldo5ggk.fsf@linutronix.de> <CANik3vFrZZTuqkht9L4b-LiP9cRWPyBDde6+4BzJVBd_xyRfoQ@mail.gmail.com> <CANik3vHWzjd+Q1NuNRB-5=OH7wPmjVBSssUPDux-eBn4i1fZcQ@mail.gmail.com> <87tvlnyob3.fsf@linutronix.de> <E19024FC-6D26-4AF2-8A91-A558C1CC05B7@converseincode.com> <CANik3vEnuBwZwOpDpurpu5kZXQWi4cES-CwKLpQwN_Jyy68uzA@mail.gmail.com> <CANik3vGQ8dRoryCMKn12WFirM1S9chM8w29nzLZOnQ1C3w-qrw@mail.gmail.com> <87va61zkq4.fsf@linutronix.de> <CANik3vFR4UeKdHuJC9By5fnQz3AsuYyRmBuhQMLC7akQ9_G9Bg@mail.gmail.com> <CANik3vG5f0pV+DVfD0zvc0vFTNb=brn26Q2SVyeKLZthBMUY1A@mail.gmail.com>
Date: Fri, 19 Oct 2018 13:51:00 +0200
In-Reply-To: <CANik3vG5f0pV+DVfD0zvc0vFTNb=brn26Q2SVyeKLZthBMUY1A@mail.gmail.com> (Beth Flanagan's message of "Wed, 17 Oct 2018 16:17:39 +0100")
Message-ID: <877eiegnob.fsf_-_@linutronix.de>
User-Agent: Gnus/5.13 (Gnus v5.13) Emacs/23.4 (gnu/linux)
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii

Specify the mcp23s18 chip so we can have a light show.

Signed-off-by: John Ogness <john.ogness@linutronix.de>
---
 arch/arm/boot/dts/am335x-pocketbeagle.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

Hi Beth,

With this change I am able to access the 16 LED's on the BaconBits
cape. Please integrate this for the ELCE. Thanks!

diff --git a/arch/arm/boot/dts/am335x-pocketbeagle.dts b/arch/arm/boot/dts/am335x-pocketbeagle.dts
index 62fe5cab9fae..8ff502a5a7dd 100644
--- a/arch/arm/boot/dts/am335x-pocketbeagle.dts
+++ b/arch/arm/boot/dts/am335x-pocketbeagle.dts
@@ -161,6 +161,25 @@
 	};
 };
 
+&spi1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi1_pins>;
+
+	mcp23s18@1 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "microchip,mcp23s18";
+		gpio-controller;
+		microchip,spi-present-mask = <0x01>;
+		reg = <1>;
+		spi-max-frequency = <1000000>;
+	};
+};
+
 &i2c2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c2_pins>;
-- 
2.11.0
